<launch>

<arg name="robot_name"/>

<arg name="robot1_init_x"/>
<arg name="robot1_init_y"/>
<arg name="robot1_init_yaw"/>

<arg name="robot2_init_x"/>
<arg name="robot2_init_y"/>
<arg name="robot2_init_yaw"/>
<arg name="map_num"/>

<group ns="/robot1/grid_map_merge">
  <param name="init_pose_x" value="$(arg robot1_init_x)"/>
  <param name="init_pose_y" value="$(arg robot1_init_y)"/>
  <param name="init_pose_z" value="0.0"/>
  <param name="init_pose_yaw" value="$(arg robot1_init_yaw)"/>
</group>

<group ns="/robot2/grid_map_merge">
  <param name="init_pose_x" value="$(arg robot2_init_x)"/>
  <param name="init_pose_y" value="$(arg robot2_init_y)"/>
  <param name="init_pose_z" value="0.0"/>
  <param name="init_pose_yaw" value="$(arg robot2_init_yaw)"/>
</group>

  <node pkg="grid_map_merge" type="grid_map_merge" respawn="false" name="grid_map_merge" output="screen">
    <param name="robot_map_topic" value="map"/>
    <param name="robot_namespace" value=""/>
    <param name="merged_map_topic" value="/$(arg robot_name)/grid_map_merge/merge_map"/>
    <param name="world_frame" value="$(arg robot_name)/merge_map"/>
    <param name="known_init_poses" value="true"/>
    <param name="merging_rate" value="2.0"/>
    <param name="discovery_rate" value="0.5"/>
    <param name="estimation_rate" value="0.5"/>
    <param name="estimation_confidence" value="1.0"/>
    <param name="map_num" value="$(arg map_num)"/>
  </node>

</launch>
